src/mujincontrollerclient.cpp

```diff
@@ -103,7 +143,13 @@ std::string ControllerClientInfo::GetURL(bool bIncludeNamePassword) const
         url += "@";
     }
 
-    url += host;
+    if (bResolveIpFromHostIfUnique) {
+        const std::string ip = ResolveIpFromHostIfUnique(host);
+        url += ip.empty() ? host : ip;
```

Comment:
If we're expecting this to guarantee uniqueness, is it correct behavior to fall back to using the `host`?